File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/APIErrorMapper.cs
using APIErrorSdk = OnlinePayments.Sdk.Domain.APIError;
using APIErrorDto = Business.Domain.Payments.PaymentDetails.APIError;

namespace Infrastructure.Mappers.PaymentDetails;

public static class ApiErrorMapper
{
    public static APIErrorDto Map(APIErrorSdk? response)
    {
        return new()
        {
            Message = response?.Message,
            ErrorCode = response?.ErrorCode,
            PropertyName = response?.PropertyName,
            HttpStatusCode = response?.HttpStatusCode,
            Retriable = response?.Retriable,
            Category = response?.Category,
            Id = response?.Id,
        };
    }
}

File: server/php/src/Infrastructure/Mappers/PaymentDetails/ApiErrorMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\APIError as APIErrorDto;
use OnlinePayments\Sdk\Domain\APIError as APIErrorSdk;

final class ApiErrorMapper
{
    public static function mapFromSdkResponse(?APIErrorSdk $response): ?APIErrorDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new APIErrorDto();
        $dto->message = $response?->message ?? null;
        $dto->errorCode = $response?->errorCode ?? null;
        $dto->propertyName = $response?->propertyName ?? null;
        $dto->httpStatusCode = $response?->httpStatusCode ?? null;
        $dto->retriable = $response?->retriable ?? null;
        $dto->category = $response?->category ?? null;
        $dto->id = $response?->id ?? null;

        return $dto;
    }
}
